docs/vaahflutter/directory_structure/root/d_lib.md

```diff
@@ -0,0 +1,47 @@
+# Lib
+
+[[toc]]
+
+## What is this directory about
+
+All your app code will reside in the lib folder. Make sure to plan your code structure before starting to develop your app. Otherwise, it can get confusing quickly without proper organization.
+
+This is the most important folder in the project, used to write most of the dart code. By default, the lib folder contains the main.dart file, which is the application’s entry point, in case of multiple environments lib can have multi main files.
+
+In our repo you can see following files and directories in lib folder
+```{20}
+project
+│   
+└─── lib
+    │
+    └─── models
+    |   | ...
+    |
+    └─── views
+    |   | ...
+    |
+    └─── controllers
+    |   | ...
+    |
+    └─── resources
+    |   | ...
+    |
+    └─── routes
+    |   | ...
+    |
+    └─── vaahextendflutter
+    |   | ...
+    |
+    └───  app_config.dart
+    |
+    └───  main.dart
+```
+
+| **Directory name** | **Documentation link** | **Description** |
+| --- | --- | --- |
+| models | [check this](../lib/models) | This directory is project specific and contains models for the project |
```

Comment:
@we-prajapati-c001 explain that to create models we're using `GetX` link it's documentation

Reply:
@themodernpk sir, on it. for models we use nothing, for controllers (state management) we are using GetX, so will describe state management and GetX.

Reply:
@themodernpk sir, In the next PR we will have the lib section https://youtu.be/Pl238bpgBrI, in that section I think we should update this section.

Reply:
For this section, if wanted we can add brief info for controllers, state management and GetX